packages/plugins/response-cache/src/redis-cache.ts

```diff
@@ -0,0 +1,115 @@
+import Redis from 'ioredis';
+import type { Cache } from './cache';
+
+export type BuildRedisEntityId = (typename: string, id: number | string) => string;
+export type BuildRedisResponseOpsKey = (responseId: string) => string;
+
+export type RedisCacheParameter = {
+  /**
+   * Redis instance
+   * @see Redis.Redis https://github.com/luin/ioredis
+   */
+  redis: Redis.Redis;
+  /**
+   * Maximum amount of items in the cache. Defaults to `Infinity`.
+   */
+  max?: number;
+  /**
+   * Customize how the cache entity id is built.
+   * By default the typename is concatenated with the id e.g. `User:1`
+   */
+  buildRedisEntityId?: BuildRedisEntityId;
+  /**
+   * Customize how the cache key that stores the operations associated with the response is built.
+   * By default `operations` is concatenated with the responseId e.g. `operations:arZm3tCKgGmpu+a5slrpSH9vjSQ=`
+   */
+  buildRedisResponseOpsKey?: BuildRedisResponseOpsKey;
+};
+
+export const createRedisCache = (params: RedisCacheParameter): Cache => {
+  const store = params.redis;
+
+  const buildRedisEntityId = params?.buildRedisEntityId ?? defaultBuildRedisEntityId;
+  const buildRedisResponseOpsKey = params?.buildRedisResponseOpsKey ?? defaultBuildRedisResponseOpsKey;
+
+  function purgeEntity(entity: string) {
+    // find the responseIds for the entity
+    store.smembers(entity).then(function (responseIds) {
+      // and purge each response since they contained the entity data
+      store.del(
+        responseIds.map(responseId => {
+          return buildRedisResponseOpsKey(responseId);
+        })
+      );
+      store.del(responseIds);
+    });
+
+    // if purging an entity like Comment, then also purge Comment:1, Comment:2, etc
+    if (!entity.includes(':')) {
+      store.keys(`${entity}:*`).then(function (entityKeys) {
+        for (const entityKey of entityKeys) {
+          // and purge any reponses in each of those entity keys
+          store.smembers(entityKey).then(function (responseIds) {
+            // if purging an entity check for associated operations containing that entity
+            store.del(
+              responseIds.map(responseId => {
+                return buildRedisResponseOpsKey(responseId);
+              })
+            );
+            // and purge each response since they contained the entity data
+            store.del(responseIds);
+          });
+        }
+
+        // then purge the entityKeys like Comment:1, Comment:2 etc
+        store.del(entityKeys);
```

Comment:
Here we could push to an array of keys as well.

Reply:
@kamilkisiela Great idea. I actually had that implemented in an earlier version and then reverted to individual commands -- but that optimization is better. I like returning a list of keys, too.

I'll make that update. Thanks!